app/jobs/characterize_job.rb

```diff
@@ -0,0 +1,12 @@
+class CharacterizeJob < ActiveJob::Base
+  queue_as :characterize
+
+  # @param [FileSet] file_set
+  # @param [String] filename a local path for the file to characterize.
+  def perform(file_set, filename)
+    Hydra::Works::CharacterizationService.run(file_set, filename)
+    file_set.mime_type = file_set.original_file.mime_type
```

Comment:
This is the only modification to characterize job. It re-sets the mime type on the FileSet because the characterization service squashes any previous value.